src/main/java/org/broadinstitute/consent/http/service/LibraryCardService.java

```diff
@@ -124,4 +128,62 @@ private void throwIfNull(LibraryCard libraryCard) {
             throw new NotFoundException("LibraryCard not found.");
         }
     }
+
+    //helper method for create method, checks to see if card already exists
+    private void checkIfCardExists(LibraryCard payload) {
+        Integer userId = payload.getUserId();
+        String email = payload.getUserEmail();
+        Integer institutionId = payload.getInstitutionId();
+        Optional<LibraryCard> foundCard;
+        List<LibraryCard> results;
+
+        if(Objects.nonNull(payload.getUserId())) {
+            results = libraryCardDAO.findLibraryCardsByUserId(userId);
+        } else if(Objects.nonNull(email)) {
+            results = libraryCardDAO.findAllLibraryCardsByUserEmail(email);
+        }
+        else {
+            throw new BadRequestException();
+        }
+        foundCard = results.stream().filter(card -> {
+            Boolean sameUserId = Objects.nonNull(userId) && card.getUserId().equals(userId);
+            Boolean sameUserEmail = Objects.nonNull(email) && card.getUserEmail().equalsIgnoreCase(email);
+            Boolean sameInstitution = card.getInstitutionId().equals(institutionId);
+            return (sameUserId || sameUserEmail) && sameInstitution;
+        }).findFirst();
+
+        if(foundCard.isPresent()) {
+            //This exception is a placeholder, what's an exception that can be translated as "CONFLICT"
+            throw new InternalServerErrorException();
+        } 
+    }
+
+    //Helper method to process user data on create LC payload
+    //Needed since CREATE has a unique situation where admins can create LCs without an active user (save with userEmail instead)
+    private LibraryCard processUserOnNewLC(LibraryCard card) {
+        if (Objects.isNull(card.getUserId())) {
+            if (Objects.isNull(card.getUserEmail())) {
+                throw new BadRequestException();
+            } else {
+                try{
+                    //If a user is found, update the card to have the correct userId associated
+                    User user = userDAO.findUserByEmail(card.getUserEmail());
+                    Integer userId = user.getDacUserId();
+                    card.setUserId(userId);
+                } catch(Exception e) {
+                    //Exception here means the email has not been used
+                    //Does not mean card is invalid, it just means it will only have user email provided
+                    //As such card can move forward as is
```

Comment:
`userDAO.findUserByEmail()` doesn't throw anything (aside from database errors), but can return a null that you'll need to handle.

This means there is no user with an email address in our system, which is fine, we don't need to have one in the user table. We can have an LC for an email that we don't know about yet.

Reply:
There's no exception thrown in the try block as described by the comments. The only exceptions you'll get are postgres ones.